plugins/woocommerce/includes/admin/class-wc-admin-notices.php

```diff
@@ -125,53 +126,72 @@ public static function reset_admin_notices() {
 	 */
 	private static function maybe_add_legacy_api_removal_notice() {
-		if ( ! self::must_show_legacy_api_removal_notice() ) {
+		if ( is_plugin_active( 'woocommerce-legacy-rest-api/woocommerce-legacy-rest-api.php' ) ) {
 			return;
 		}
 
-		self::add_custom_notice(
-			'legacy_api_removed_in_woo_90',
-			sprintf(
-				'%s%s',
+		if ( 'yes' === get_option( 'woocommerce_api_enabled' ) ) {
+			self::add_custom_notice(
+				'legacy_api_removed_in_woo_90',
 				sprintf(
-					'<h4>%s</h4>',
-					esc_html__( 'The WooCommerce Legacy REST API will be removed soon', 'woocommerce' )
-				),
+					'%s%s',
+					sprintf(
+						'<h4>%s</h4>',
+						esc_html__( 'The WooCommerce Legacy REST API will be removed soon', 'woocommerce' )
+					),
+					sprintf(
+					// translators: Placeholders are URLs.
+						wpautop( wp_kses_data( __( 'The WooCommerce Legacy REST API, <a href="%1$s">currently enabled in this site</a>, will be removed in WooCommerce 9.0. A separate WooCommerce extension will be available to keep it enabled. <b><a target=”_blank” href="%2$s">Learn more about this change.</a></b>', 'woocommerce' ) ) ),
+						admin_url( 'admin.php?page=wc-settings&tab=advanced&section=legacy_api' ),
+						'https://developer.woocommerce.com/2023/10/03/the-legacy-rest-api-will-move-to-a-dedicated-extension-in-woocommerce-9-0/'
+					)
+				)
+			);
+		}
+
+		if ( wc_get_container()->get( WebhookUtil::class )->get_legacy_webhooks_count() > 0 ) {
+			self::add_custom_notice(
+				'legacy_webhooks_unsupported_in_woo_90',
 				sprintf(
+					'%s%s',
+					sprintf(
+						'<h4>%s</h4>',
+						esc_html__( 'The WooCommerce webhooks with Legacy REST API payload will be unsupported soon', 'woocommerce' )
+					),
+					sprintf(
 					// translators: Placeholders are URLs.
-					wpautop( wp_kses_data( __( 'The WooCommerce Legacy REST API, <a href="%1$s">currently enabled in this site</a>, will be removed in WooCommerce 9.0. A separate WooCommerce extension will be available to keep it enabled. <b><a target=”_blank” href="%2$s">Learn more about this change.</a></b>', 'woocommerce' ) ) ),
-					admin_url( 'admin.php?page=wc-settings&tab=advanced&section=legacy_api' ),
-					'https://developer.woocommerce.com/2023/10/03/the-legacy-rest-api-will-move-to-a-dedicated-extension-in-woocommerce-9-0/'
+						wpautop( wp_kses_data( __( 'The WooCommerce Legacy REST API will be removed in WooCommerce 9.0, and this will cause <a href="%1$s">the Legacy REST API payload webhooks currently existing in this site</a> to stop working. A separate WooCommerce extension will be available to keep the Legacy REST API enabled and these webhooks working, you can also edit these webhooks to use the current REST API to generate the payload. <b><a target=”_blank” href="%2$s">Learn more about this change.</a></b>', 'woocommerce' ) ) ),
```

Comment:
```suggestion
						wpautop( wp_kses_data( __( 'The WooCommerce Legacy REST API will be removed in WooCommerce 9.0, and this will cause <a href="%1$s">webhooks on this site that are configured to use the Legacy REST API</a> to stop working. A separate WooCommerce extension will be available to allow these webhooks to keep using the Legacy REST API without interruption. You can also edit these webhooks to use the current REST API version to generate the payload instead. <b><a target=”_blank” href="%2$s">Learn more about this change.</a></b>', 'woocommerce' ) ) ),
```